### PR TITLE
prefix k for constants in naming rules

### PR DESCRIPTION
**Pull request by slava77**
we do not have a style for constant names. It would be useful to have something.

This is not supposed to refer to regular values which happen to be const, but rather to something that can be used as `name::kConstant`

### CODE REVIEW

**Comment by makortel** [comment]
Thanks Slava, I suppose we should discuss this in the next core meeting.

**Comment by slava77** [comment]
> > Thanks Slava, I suppose we should discuss this in the next core meeting.
> 
> sure

I thought there would be one today.
Next week is the O&C week.
Is it two more weeks then?

File: cms_coding_rules.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ Coding rules are meant to prevent serious problems in software function, perform
 14. Do not use "\_" as ﬁrst character, except for user-defined suffixes (used in user-defined literals). Only use it as the last character for class data member names, not local variable names.
 15. Do not use "\_\_".
 16. Use clear and explanatory variable names.
+17. For class/enum/struct/namespace/global scope constants prefix `k` is slightly preferred (this applies to what can be referred to as `TypeName::kConstantName` or `namespaceName::kConstantName`) . (\*)
```
Comment (makortel) [inline cms_coding_rules.md:45]: From core sw meeting something along
```suggestion
17. For class/enum/struct/namespace/global scope constants prefix `k` is preferred for new sets of constants (this applies to what can be referred to as `TypeName::kConstantName` or `namespaceName::kConstantName`) . (\*) Allowed exception: existing sets of constants can continue to use different naming convention,
```

**Comment by smuzaffar** [comment]
@makortel , as discussed I am merging this

**Pull request merged**